File: service/src/test/java/no/nav/melosys/service/soknad/SoknadMottattTest.java
package no.nav.melosys.service.soknad;

import java.time.ZonedDateTime;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class SoknadMottattTest {

    @Test
    void mindreEnn7dagerSidenMottak_erIkkeGammel() {
        SoknadMottatt soknadMottatt = new SoknadMottatt("ID", ZonedDateTime.now().minusDays(6).minusHours(23));

        assertThat(soknadMottatt.erForGammelTilForvaltningsmelding()).isFalse();
    }

    @Test
    void merEnn7dagerSidenMottak_erGammel() {
        SoknadMottatt soknadMottatt = new SoknadMottatt("ID", ZonedDateTime.now().minusDays(7).minusHours(1));

        assertThat(soknadMottatt.erForGammelTilForvaltningsmelding()).isTrue();
    }
}

File: service/src/test/kotlin/no/nav/melosys/service/soknad/SoknadMottattTest.kt
package no.nav.melosys.service.soknad

import io.kotest.matchers.shouldBe
import no.nav.melosys.integrasjon.SoknadMottatt
import org.junit.jupiter.api.Test
import java.time.ZonedDateTime

class SoknadMottattTest {

    @Test
    fun `mindreEnn7dagerSidenMottak skal returnere false for erForGammelTilForvaltningsmelding`() {
        val soknadMottatt = SoknadMottatt("ID", ZonedDateTime.now().minusDays(6).minusHours(23))

        soknadMottatt.erForGammelTilForvaltningsmelding() shouldBe false
    }

    @Test
    fun `merEnn7dagerSidenMottak skal returnere true for erForGammelTilForvaltningsmelding`() {
        val soknadMottatt = SoknadMottatt("ID", ZonedDateTime.now().minusDays(7).minusHours(1))

        soknadMottatt.erForGammelTilForvaltningsmelding() shouldBe true
    }
} 